packages/katello/katello/katello.spec

```diff
@@ -67,8 +67,9 @@ Requires(post): candlepin >= 2.0
 Requires: candlepin-selinux >= 2.0
 Requires: java-openjdk >= 1:1.7.0
 Requires: java-openjdk < 1:1.8.0.45
-Requires: postgresql
-Requires: postgresql-server
+Requires: /usr/bin/psql
+Requires: /usr/bin/pg_dump
+Requires: /usr/bin/pg_dumpall
```

Comment:
There's no just client package that can be required?

Reply:
I think this enabled the postgresql 9.6 SCL to be used since there's a package that provides the binaries in those locations.

Reply:
But the native `postgresl` package does not require them? This confuses me a bit cause how do we know which postgresl packages we are going to get if this provides is met by the non-scl, and all of the SCLs?

`postgresql` doesn't translate to `rh-postgresql96-postgresql-9.6.5-1.el7.x86_64.rpm` ?

Reply:
From a scripting perspective we're happy with any /usr/bin/psql we can get. I am not 100% sure which pkg will be installed when multiple provide the same path.

Reply:
Can we test the latter part to see what version of postgresl packages get installed if the SCL is enabled?

Reply:
so I did not test the RPM itself (will do tomorrow), but plain `yum` does what we would like to see:

```
[root@centos7 ~]# yum install /usr/bin/psql
Loaded plugins: fastestmirror
base                                                                                                                                                                                                                                                                                                   | 3.6 kB  00:00:00     
extras                                                                                                                                                                                                                                                                                                 | 3.4 kB  00:00:00     
updates                                                                                                                                                                                                                                                                                                | 3.4 kB  00:00:00     
(1/4): base/7/x86_64/group_gz                                                                       | 156 kB  00:00:00     
(2/4): extras/7/x86_64/primary_db                                                                   | 166 kB  00:00:00     
(3/4): base/7/x86_64/primary_db                                                                     | 5.7 MB  00:00:01     
(4/4): updates/7/x86_64/primary_db                                                                  | 6.0 MB  00:00:06     
Determining fastest mirrors                                       
 * base: mirror.418grad.media
 * extras: mirror.418grad.media
 * updates: ftp.fau.de
Resolving Dependencies
--> Running transaction check
---> Package postgresql.x86_64 0:9.2.23-3.el7_4 will be installed
--> Processing Dependency: postgresql-libs(x86-64) = 9.2.23-3.el7_4 for package: postgresql-9.2.23-3.el7_4.x86_64
--> Processing Dependency: libpq.so.5()(64bit) for package: postgresql-9.2.23-3.el7_4.x86_64
--> Running transaction check
---> Package postgresql-libs.x86_64 0:9.2.23-3.el7_4 will be installed
--> Finished Dependency Resolution

Dependencies Resolved

======================================================================================================================
 Package                              Arch         Version            Repository     Size
======================================================================================================================
Installing:
 postgresql                           x86_64       9.2.23-3.el7_4     updates        3.0 M
Installing for dependencies:
 postgresql-libs                      x86_64       9.2.23-3.el7_4     updates        234 k

Transaction Summary
======================================================================================================================
Install  1 Package (+1 Dependent package)

Total download size: 3.3 M
Installed size: 17 M
Is this ok [y/d/N]: n

[root@centos7 ~]# yum install -y centos-release-scl

[root@centos7 ~]# yum install /usr/bin/psql
Loaded plugins: fastestmirror
centos-sclo-rh                                                                                     | 2.9 kB  00:00:00     
centos-sclo-sclo                                                                                   | 2.9 kB  00:00:00     
(1/2): centos-sclo-sclo/x86_64/primary_db                                                          | 196 kB  00:00:01     
(2/2): centos-sclo-rh/x86_64/primary_db                                                            | 3.0 MB  00:00:31     
Loading mirror speeds from cached hostfile
 * base: mirror.418grad.media
 * extras: mirror.418grad.media
 * updates: ftp.fau.de
Resolving Dependencies
--> Running transaction check
---> Package postgresql.x86_64 0:9.2.23-3.el7_4 will be installed
--> Processing Dependency: postgresql-libs(x86-64) = 9.2.23-3.el7_4 for package: postgresql-9.2.23-3.el7_4.x86_64
--> Processing Dependency: libpq.so.5()(64bit) for package: postgresql-9.2.23-3.el7_4.x86_64
--> Running transaction check
---> Package postgresql-libs.x86_64 0:9.2.23-3.el7_4 will be installed
--> Finished Dependency Resolution

Dependencies Resolved

======================================================================================================================
 Package                              Arch         Version            Repository     Size
======================================================================================================================
Installing:
 postgresql                           x86_64       9.2.23-3.el7_4     updates        3.0 M
Installing for dependencies:
 postgresql-libs                      x86_64       9.2.23-3.el7_4     updates        234 k

Transaction Summary
======================================================================================================================
Install  1 Package (+1 Dependent package)

Total download size: 3.3 M
Installed size: 17 M

[root@centos7 ~]# yum install -y rh-postgresql96-postgresql-syspaths.x86_64

[root@centos7 ~]# yum install /usr/bin/psql
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: mirror.418grad.media
 * extras: mirror.418grad.media
 * updates: ftp.fau.de
Package rh-postgresql96-postgresql-syspaths-9.6.5-1.el7.x86_64 already installed and latest version
Nothing to do
```